lnwallet/wallet.go

```diff
@@ -921,6 +868,74 @@ func (l *LightningWallet) handleSingleContribution(req *addSingleContributionMsg
 	return
 }
 
+// createAndSignFundingTx is called to create a funding transaction for a
+// pending reservation using the given multi-sig output. It will populate
+// the fundingTx and ourFundingInputScripts fields of the reservation.
+func (l *LightningWallet) createAndSignFundingTx(
+	reservation *ChannelReservation, multiSigOut *wire.TxOut) error {
+
+	reservation.fundingTx = wire.NewMsgTx(1)
+	fundingTx := reservation.fundingTx
+
+	ourContribution := reservation.ourContribution
```

Comment:
Style suggestion: to avoid mutating passed arguments, you can make the method take `our|theirContribution` and return the `fundingTx` + `ourFundingInputScripts`.

Reply:
That's a great idea. I just pushed a new commit addressing your suggestion.